### ISSUE TITLE
Frontend Progress

### ISSUE THREAD

**Issue by emmaxshen** [e-shen2022/emma_blog#49]
***Read comments for in-depth line by line explanation***

1. Import Packages 
<img width="254" alt="image" src="https://github.com/e-shen2022/APCSP_Blog/assets/104966589/b684968b-e3b2-4a3a-b53b-bf8d46f7194b">

- OpenCV to open camera and capture real-time webcam frames
- Threading & events to coordinate subject and avatar frames 

2. Generate path and create dictionaries for emotion text and avatar images
<img width="438" alt="image" src="https://github.com/e-shen2022/APCSP_Blog/assets/104966589/6e5edc40-44d3-4d57-a8c0-be87aa27f56a">

3. Defining/Initializing variables
<img width="375" alt="image" src="https://github.com/e-shen2022/APCSP_Blog/assets/104966589/501632fe-5eee-4022-986f-0b3a77394a9a">

- Define variable for frame captured
- Initialize array to hold index of emotion detected
- Switch event to signal when one of functions finish and should start next
- Stop event used when quit button is pressed

4. Show_subject function
<img width="577" alt="image" src="https://github.com/e-shen2022/APCSP_Blog/assets/104966589/fa7327e0-8aae-4856-acc7-9a8aa59ff225">
<img width="644" alt="image" src="https://github.com/e-shen2022/APCSP_Blog/assets/104966589/a5f6b8b4-ea67-4598-a79b-16406cd495ef">
<img width="464" alt="image" src="https://github.com/e-shen2022/APCSP_Blog/assets/104966589/68e65199-333d-4d74-b326-638a550f4c3e">

- While program is running, open the webcam and capture the frame
- Use haarcascade classifier that detects face in frame using pretrained info ab facial features
- For each detected face, a rectangle is drawn around it on the frame
- Face image is passed through the pre-trained emotion recognition model 
- Index of the highest probability in the prediction array is calculated to determine the predicted emotion class
- Corresponding emotion label is retrieved from the emotion_dict dictionary & displayed in window
- Once frame is processed, wait for 0.3 seconds to account for lag 

5. Show_avatar function
<img width="488" alt="image" src="https://github.com/e-shen2022/APCSP_Blog/assets/104966589/c44730e2-f7aa-49f6-a582-b8e4213b94bd">

- While quit button is not pressed, read the frame, display same emotion text as subject

6. Wrapper functions
<img width="479" alt="image" src="https://github.com/e-shen2022/APCSP_Blog/assets/104966589/d7f69972-394e-4688-bae5-1794c8ec1ee6">

- Run stop_thread function when quit button is pressed
- Wrapper_quit to stop all threads and destroy root 

7. Main commands
<img width="643" alt="image" src="https://github.com/e-shen2022/APCSP_Blog/assets/104966589/f69bf612-3b5a-49cf-90a8-5644361fce6f">

- Create labels to position and pack the human subject, emoji, and buttons
